### DISCUSSION TITLE
Sending 0xC0 and 0xC1 (ON and OFF) commands to my JVC AMP through its CompuLink jack

### DISCUSSION THREAD

**Discussion by MMMXXXZZZ** [sle118/squeezelite-esp32#407] [Q&A]
The old JVCs have a 3.5mm jack that receives CompuLink commands (they are in binary). Is there an addon for Squeezelite to send timed binary commands through GPIO?

With almost no programming experience I understand that a GPIO pin could connect to the CompuLink jack in the AMP and send the binary data according to the guide below, where somebody decoded the commands.

I assume powering the ESP32 with 3.3V means GPIO will output 3.3V. Might not be true but I'm very comfortable figuring out how to step up the voltage if the receiver doesn't receive below 5v. It's just the code that I have no idea about.

If there's anyone with some experience out there that can help me out reaching the last mile of this project I'd really appreciate it.

The intended behavior is for the ESP32 to be active regardless. On playback, send the ON command, and on 15 minutes of no playback send the OFF command. SqueezeLite must be able to be accessed, connected to, etc, (i.e. always on) regardless of the AMP being off.

I don't know how addons are added into SqueezeLite, I'd be willing to do anything, even learn to compile, although I'm more familiar with the good old drag and drop file into addons folder-type process.

_From https://github.com/jcj83429/jvc_compulink_:
## Data Encoding
Each word contains 8 bits, transmitted from MSB to LSB. (The service manual of the SEA-RM20 remote control unit lists the command values, and I observed the commands being transmitted from MSB to LSB) The bit encoding is similar to the [NEC remote protocol](https://www.circuitvalley.com/2013/09/nec-protocol-ir-infrared-remote-control.html):

        0    1
    5V __  __
    
    0V   __  ______
       \  /\      /      
       10ms  20ms
- A "0" bit is encoded as **5ms of 5V followed by 5ms of 0V.**
- A "1" bit is encoded as **5ms of 5V followed by 15ms of 0V.**
- The word is ended with 5ms of 5V to allow the receiver to tell if the last bit is 0 or 1.

Example: 0x20 on the wire:

        0   0     1     0   0   0   0   0  END
    5V __  __  __      __  __  __  __  __  __
    
    0V   __  __  ______  __  __  __  __  __  

### Tuner
COMMAND   | VALUE
----------|------
POWER ON  | 0xC0
POWER OFF | 0xC1

**Comment by sle118** [comment]
If anyone is willing to try an implementation, I'd be glad to help with merging. Aside from implementing in a custom branch, this will not happen in the current state of the system. We have limited capacity and @philippe44 is working hard on keeping the ecosystem alive, given that Logitech had decided to lower the expense on the platform (for an incredible amount of time after they made their last hardware device).

You could also try to implement infrared control. Not perfect but it might be good enough. 

In the meantime, and for the reference: 
https://github.com/sle118/squeezelite-esp32/discussions/342
